### DISCUSSION TITLE
MQTT not working

### DISCUSSION THREAD

**Discussion by fcrozat** [seaky/nspanel_pro_tools_apk#81] [Q&A]
I switch to version 2.2 and tried to enable MQTT support but so far, it doesn't seem able to connect to my MQTT broker. Moreover, when I try to change some of the MQTT settings, such as port, IP, application has a tendency to crash.

**Comment by seaky** [comment]
What do you mean version 2.2? Is this the firmware version?

If you still have crash please in App -> Settings -> Debug category enable Debug mode and Verbose mode and after crash please post here the log.

Thanks in advance.

**Reply by fcrozat** [reply]
I've filled https://github.com/seaky/nspanel_pro_tools_apk/issues/82 for the crash.
System is running firmware 2.3 and nspanel_pro_tools 2.2

**Reply by seaky** [reply]
Are you sure you'are using the nspanel pro tools v2.2? As far as I know, the last release I made was v2.1.0 :)

**Reply by fcrozat** [reply]
My bad, between nspanel pro tools, firmware and the "official" sonoff app version, I mixed things ;)

So, indeed, it is v2.1 of your app.

I didn't fill username because my MQTT broker is running in anonmymous mode, so it didn't feel needed.

I just did (no password filled) and it was able to connect. Thanks

**Comment by seaky** [comment]
Getting logs from logcat:

1.) Enable Debug and Verbose log
2.) get pid while program is running: ```adb shell ps -Ao PID,NAME | grep com.seaky.nspanelpro.tools```
after crash
3.) create log file:  ```adb logcat --pid=<pid> -d  > crash.log```

**Comment by seaky** [comment]
Issue was handled here https://github.com/seaky/nspanel_pro_tools_apk/issues/82
